src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+    redownload: bool = False,
+    start_dekad: int = 10,
+):
+    """Process DMP raw data
+
+    DMP raw data is received in a wide format
+    dataset that needs processing. This pivots
+    the data from wide to long format with
+    DMP dekadal observations for all years, while
+    also pivoting the mean values to allow for
+    calculation of biomasse anomaly.
+    """
+    df = load_dmp(admin, redownload)
+
+    # process mean and DMP separately since mean values
+    # are dekadal and DMP are year/dekadal
+    # keep ID column for working with multipolygon admin areas
+    df_mean = df.filter(regex="(^admin|^DMP_MEA|^ID)")
+    df_dmp = df.filter(regex="(^admin|^DMP_[0-9]+|^ID)")
+    admin_cols = [col for col in df_mean.columns if col.startswith("admin")]
+
+    # groupby to average out for the few cases where admin areas
+    # are not contiguous
+    df_mean_long = (
+        pd.wide_to_long(
+            df_mean,
+            i=admin_cols + ["IDBIOHYDRO"],
+            j="dekad",
+            stubnames="DMP_MEA",
+            sep="_",
+        )
+        .reset_index()
+        .drop(labels="IDBIOHYDRO", axis=1)
+        .groupby(admin_cols + ["dekad"])
+        .mean()
+        .reset_index()
+    )
+
+    # calculate anomaly for mean separate from
+    # observed to ensure unique dekadal values
+    df_mean_long["season_index"] = np.where(
+        df_mean_long["dekad"] >= start_dekad,
+        df_mean_long["dekad"] - start_dekad,
+        df_mean_long["dekad"] + (36 - start_dekad),
+    )
+    df_mean_long.sort_values(by=admin_cols + ["season_index"], inplace=True)
+    df_mean_long["biomasse_mean"] = df_mean_long.groupby(by=admin_cols)[
+        "DMP_MEA"
+    ].apply(lambda x: x.cumsum() * 365.25 / 36)
+
+    df_dmp_long = (
+        pd.wide_to_long(
+            df_dmp,
+            i=admin_cols + ["IDBIOHYDRO"],
+            j="time",
+            stubnames="DMP",
+            sep="_",
+        )
+        .reset_index()
+        .drop(labels="IDBIOHYDRO", axis=1)
+        .groupby(admin_cols + ["time"])
+        .sum()
+        .reset_index()
+    )
+
+    # convert time into year and dekad
+    df_dmp_long[["year", "dekad"]] = (
+        df_dmp_long["time"].apply(str).str.extract("([0-9]{4})([0-9]{2})")
+    )
+    df_dmp_long = df_dmp_long.astype({"year": int, "dekad": int})
+
+    # drop values from 1999 that are not from a complete season
+    df_dmp_long = df_dmp_long[
+        ~((df_dmp_long["year"] == 1999) & (df_dmp_long["dekad"] < start_dekad))
+    ]
+
+    # join mean and observed values
+    df_merged = pd.merge(
+        left=df_dmp_long, right=df_mean_long, on=admin_cols + ["dekad"]
+    )
+
+    # calculate biomasse (cumulative sum of DMP across the season
+    # starting with the start_dekad and ending at start_dekad - 1
+    df_merged.sort_values(
+        by=admin_cols + ["year", "dekad"], inplace=True, ignore_index=True
+    )
+    df_merged["season"] = df_merged["year"] + (
```

Comment:
I am a bit confused by the definition of season, I would expect it to be one lower.. Cause now e.g. the 2022 season starts in the `start_dekad` of 2021, so on 1 April 2021. It is not super bad but would at least add a comment explaining that. And be careful to later on not mix up season definition

Reply:
Yeah, I will subtract from 1 so it's clear, but just added it here for the grouping, it wasn't actually included in the output file. However, have decided probably good to add to the output as well, so will make it even clearer by having it be "year1-year2".